### DISCUSSION TITLE
WiFiEspAT lib possibly locking up USB HID communication?

### DISCUSSION THREAD

**Discussion by FeuerSturm** [JAndrassy/WiFiEspAT#82] [Q&A]
Hey guys,
I'm having a strange problem with the iLabs Challenger 2040 WiFi MCU (RP2040 based MCU with embedded ESP8285 chip):
I'm using TinyUSB to create a generic USB HID device that listens to usb packets, reformats them and sends them via UDP to a receiver (WLED flashed ESP32).
This works perfect for an Arduino Nano RP2040 Connect but with the "same" sketch that is just modified to work with said Challenger MCU (WiFiEspAT instead of WiFiNINA lib) it is "blocking" receiving the packets via usb and I am receiving "Overlapping I/O"-errors when trying to send usb hid packets to it with the App I am using.
Anyone ever had similar problems?

This is the sketch in question, maybe one of the pros sees a blatant mistake, I've been staring at it for hours and just don't see what could be wrong. Thanks in advance!
[SRGBmods_Wifi_LED_Controller_TX_BETA2.txt](https://github.com/JAndrassy/WiFiEspAT/files/8667916/SRGBmods_Wifi_LED_Controller_TX_BETA2.txt)
!

**Comment by JAndrassy** [comment]
you could try much higher AT_BAUD_RATE so the Serial communication is faster (the WiFiEspAT library blocks while waiting for response from the AT firmware). don't forget to change the baud rate of AT firmware.

**Reply by FeuerSturm** [reply]
Thanks for the really fast reply!

I've tried with a baud rate of 921600 already, unfortunately that didn't change anything.
I even tried 2000000 without any luck.

Would you have any more ideas?

**Comment by JAndrassy** [comment]
only system resource used by WiFiEspAT is the Serial (UART). so only possible resource conflicts are UART interrupts or busy CPU. 
a.f.a.i.k. TinyUSB is a bit-banged USB so it excessively uses CPU (and interrupts maybe). 
RP2040 has USB peripheral so maybe use it instead of TinyUSB.

**Reply by FeuerSturm** [reply]
Thanks for your reply,
I'll play around with it a bit and try Pico-SDK instead of TinyUSB even though I am not sure all the features I need will be supported.
